http2_general/test_h2_responses.py

```diff
@@ -116,3 +120,219 @@ def test_h2_bad_forwarded_for_ip(self):
+
+        i = 0
+        for id in self.clients_ids:
+            client = self.get_client(id)
+            i = i + 1
+            if i == bad_num:
+                self.assertEqual(client._last_response.status, "502")
+            elif i > bad_num:
+                self.assertFalse(client._last_response)
+            else:
+                self.assertTrue(client.wait_for_response())
+                self.assertEqual(client._last_response.status, "200")
+
+        srv.wait_for_connections()
+        req_count = i
+
+        i = 0
+        j = 0
+        for id in self.clients_ids:
+            i = i + 1
+            client = self.get_client(id)
+            if i > bad_num:
+                j = j + 1
+                self.assertTrue(srv.wait_for_requests(req_count + j))
+                srv.flush()
+                self.assertTrue(client.wait_for_response())
+                self.assertEqual(client._last_response.status, "200")
+
+
+class H2HmResponsesPipelined(H2ResponsesPipelinedBase):
+    tempesta = {
+        "config": """
+            listen 443 proto=h2;
+            frang_limits {
+                http_strict_host_checking false;
+            }
+            
+            health_check hm0 {
+                request         "GET / HTTP/1.0\r\n\r\n";
+                request_url     "/";
+                resp_code       200;
+                resp_crc32  0x31f37e9f;
+                timeout         2;
+            }
+
+            srv_group default {
+                server ${server_ip}:8000 conns_n=1;
+                health hm0;
+            }
+            vhost good {
+                proxy_pass default;
+            }
+            tls_certificate ${tempesta_workdir}/tempesta.crt;
+            tls_certificate_key ${tempesta_workdir}/tempesta.key;
+            tls_match_any_server_name;
+            http_max_header_list_size 134217728; #128 KB
+
+            block_action attack reply;
+            block_action error reply;
+            http_chain {
+                host == "bad.com"   -> block;
+                                    -> good;
+            }
+        """
+    }
+
+    clients_ids = ["deproxy_1", "deproxy_2", "deproxy_3"]
+
+    @parameterize.expand(
+        [
+            param(name="1_hm", hm_num=1),
+            param(name="2_hm", hm_num=2),
+            param(name="3_hm", hm_num=3),
+            param(name="4_hm", hm_num=4),
+        ]
+    )
+    def test_hm_pipelined(self, name, hm_num):
+        srv = self.setup_and_start(4)
+        self.disable_deproxy_auto_parser()
+
+        srv.set_response(
+            "HTTP/1.1 200 OK\r\n"
+            + f"Date: {HttpMessage.date_time_string()}\r\n"
+            + "Server: debian\r\n"
+            + "Content-Length: 0\r\n\r\n"
+        )
+
+        i = 0
+        for id in self.clients_ids:
+            client = self.get_client(id)
+            i = i + 1
+            if i == hm_num:
+                self.assertTrue(srv.wait_for_requests(i))
```

Comment:
Please add comment. This should help others understand the test better.
```suggestion
                self.assertTrue(srv.wait_for_requests(i), "Server did not receive hm request from TempestaFW.")
```